src/test/java/org/apache/pulsar/io/jcloud/format/AvroFormatTest.java

```diff
@@ -63,13 +65,17 @@ public void handleMessage(TopicName topicName, Message<GenericRecord> msg) {
         List<Record<GenericRecord>> records = new ArrayList<>();
         records.add(new SinkRecord<>(test, test.getValue()));
         try {
-            final Schema<GenericRecord> schema = AvroRecordUtil.extractPulsarSchema(msg);
+            final Schema<GenericRecord> schema = (Schema<GenericRecord>) msg.getReaderSchema().get();
+            final BlobStoreAbstractConfig config = new BlobStoreAbstractConfig();
+            config.setUseMetadata(true);
+            ((InitConfiguration<BlobStoreAbstractConfig>) getFormat()).configure(config);
             getFormat().initSchema(schema);
             ByteSource byteSource = getFormat().recordWriter(records.listIterator());
 
 
+            final org.apache.avro.Schema avroSchema = AvroRecordUtil.convertToAvroSchema(schema);
             final GenericDatumReader<Object> datumReader =
-                    new GenericDatumReader<>(AvroRecordUtil.convertToAvroSchema(schema));
+                    new GenericDatumReader<>(MetadataUtil.setMetadataSchema(avroSchema));
```

Comment:
Do we have a test to cover the new change? I can't find any tests for verifying if the data contains the message metadata